spinel/stream.py

```diff
@@ -55,8 +55,9 @@ class StreamSerial(IStream):
     def __init__(self, dev, baudrate=115200):
         try:
             self.serial = serial.Serial(dev, baudrate)
-        except:
-            CONFIG.LOGGER.error("Couldn't open " + dev)
+        except Exception as e:
+            CONFIG.LOGGER.error(f"Couldn't open {dev} {str(e)}")
+            self.close()
```

Comment:
If exception is raised here, `self.serial` is not assigned correctly, calling `self.close` => `self.serial.close()` will raise `AttributeError`

Reply:
Updated.